utils.h

```diff
@@ -0,0 +1,9 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#include "mkmimo.h"
```

Comment:
Let's turn utils.[ch] into buffer.[ch] and keep things basic here, avoiding including mkmimo.h.